coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/TrinoSqlUDFConverter.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright 2017-2023 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.trino.rel2trino;
+
+import java.util.Map;
+
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.util.SqlShuttle;
+
+import static com.linkedin.coral.trino.rel2trino.CoralTrinoConfigKeys.*;
+
+
+/**
+ * This class extends the class of SqlShuttle and calls CalciteTrinoUDFOperatorTransformers to traverse
+ * the hierarchy and converts UDF operator in all SqlCalls if it is required
+ */
+public class TrinoSqlUDFConverter extends SqlShuttle {
+  private static final String TO_DATE_OPERATOR_NAME = "to_date";
+  private final Map<String, Boolean> configs;
+  public TrinoSqlUDFConverter(Map<String, Boolean> configs) {
+    this.configs = configs;
+  }
+
+  @Override
+  public SqlNode visit(SqlCall call) {
+    if (shouldTransformOperator(call.getOperator().getName())) {
+      CalciteTrinoUDFOperatorTransformers operatorTransformers = CalciteTrinoUDFOperatorTransformers.getInstance();
+      SqlCall transformedCall = operatorTransformers.apply(call);
+      return super.visit(transformedCall);
+    } else {
+      return super.visit(call);
+    }
+  }
+
+  private boolean shouldTransformOperator(String operatorName) {
```

Comment:
Why is not implemented as an ad hoc transformer?

Reply:
Good idea. I have move this condition and related transformation into a new SqlCallTransformer class named `ToDateOperatorTransformer`